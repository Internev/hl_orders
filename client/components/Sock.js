import React from 'react'

const Sock = ({sock, handleFormSubmit}) => (
  <div className='sock-tile'>
    <div className='sock-tile-head'>
      <div className='sock-tile-title'>
        {sock.styleID}
      </div>
      <div>
        <i>{sock.desc}</i>
      </div>
    </div>
    <form className='sock-tile-form'>
      <table>
        <tbody>
          <tr>
            <th>Colour ID</th>
            <th>Colour Name</th>
            <th>Pattern</th>
            {sock.sizes.map(size => (
              <th key={size}>{size}</th>
            ))}
          </tr>
          {sock.colours.map((colour, index) => (
            <tr key={index}>
              <td>{colour.colourID}</td>
              <td>{colour.colourName}</td>
              {colour.patternName !== 'NONE'
                ? <td>{colour.patternID} - {colour.patternName}</td> : <td />}
              {sock.sizes.map(size => (
                colour.hasOwnProperty(size)
                ? <td key={size}>
                  <input
                    className='order-input'
                    type='number'
                    min='0'
                    value={colour[size] === 0 ? '' : colour[size]}
                    onChange={(e) => handleFormSubmit(sock, colour, e.target.value, size, index)}
                    />
                </td>
                : <td key={size} />
            ))}
            </tr>
        ))}
          {sock.totalAmt > 0
          ? <tr>
            <td colSpan={2} style={{ borderTop: '1px solid grey' }}>Subtotal:</td>
            <td style={{ borderTop: '1px solid grey' }} />
            {sock.sizes.map(size => (
              <td key={size} style={{ textAlign: 'center', borderTop: '1px solid grey' }}>{sock.colours.reduce((memo, colour) => {
                if (colour.hasOwnProperty(size)) memo += colour[size]
                return memo
              }, 0)}</td>
            ))}
          </tr>
          : null
          }
        </tbody>
      </table>
    </form>
    <div>{sock.totalAmt ? sock.totalAmt : '0'} Total pairs @ ${sock.price.toFixed(2)} exGST: ${sock.totalAmt
        ? (sock.price * sock.totalAmt).toFixed(2) : '0.00'}</div>
  </div>
)

export default Sock
