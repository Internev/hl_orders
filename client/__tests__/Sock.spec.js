import React from 'react'
import {shallow} from 'enzyme'
import Sock from '../components/Sock'

describe('Sock Display Component', () => {
  const colour1 = {colourID: '74', colourName: 'ANTELOPE', patternID: '0', patternName: 'NONE', '01C05': 0, '01C07': 0, '01C10': 0}
  const colour2 = {colourID: '66', colourName: 'DENIM', patternID: '0', patternName: 'NONE', '01C05': 0, '01C07': 0, '01C10': 0}
  const socky = {styleID: '01C', desc: 'Alpaca Health SockÂ®', price: 13.75, totalAmt: 0, colours: [colour1, colour2], sizes: ['01C05', '01C07', '01C10']}
  let wrapper

  beforeEach(() => {
    wrapper = shallow(
      <Sock sock={socky} handleFormSubmit={()=>{}} />
    )
  })

  it('should render', () => {
    expect(wrapper.find('input').exists()).toBe(true)
    expect(wrapper.find('table').exists()).toBe(true)
  })

  it('should have 2 colours (and header row)', () => {
    expect(wrapper.find('tbody > tr').map(i => i.html())).toMatchSnapshot()
  })

  it('should render the colour names', () => {
    socky.colours.forEach(colour => {
      expect(
        wrapper.containsMatchingElement(
          <td>{colour.colourName}</td>
        )
      ).toBe(true)
    })
  })
})
