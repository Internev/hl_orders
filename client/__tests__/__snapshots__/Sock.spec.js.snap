// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Sock Display Component should have 2 colours (and header row) 1`] = `
Array [
  "<tr><th>Colour ID</th><th>Colour Name</th><th>Pattern</th><th>01C05</th><th>01C07</th><th>01C10</th></tr>",
  "<tr><td>74</td><td>ANTELOPE</td><td></td><td><input type=\\"number\\" min=\\"0\\" class=\\"order-input\\" value=\\"\\"/></td><td><input type=\\"number\\" min=\\"0\\" class=\\"order-input\\" value=\\"\\"/></td><td><input type=\\"number\\" min=\\"0\\" class=\\"order-input\\" value=\\"\\"/></td></tr>",
  "<tr><td>66</td><td>DENIM</td><td></td><td><input type=\\"number\\" min=\\"0\\" class=\\"order-input\\" value=\\"\\"/></td><td><input type=\\"number\\" min=\\"0\\" class=\\"order-input\\" value=\\"\\"/></td><td><input type=\\"number\\" min=\\"0\\" class=\\"order-input\\" value=\\"\\"/></td></tr>",
]
`;
